src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -113,6 +113,13 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     def knownDirectSubclasses = {
       // See `getFlag` to learn more about the `isThreadsafe` call in the body of this method.
       if (!isCompilerUniverse && !isThreadsafe(purpose = AllOps)) initialize
+
+      enclosingPackage.info.decls.foreach { sym =>
+        if(sourceFile == sym.sourceFile) {
```

Comment:
The sealedness constraint means that we only need to search within the same compilation unit.